test/test_autograd.py

```diff
@@ -5037,7 +5037,7 @@ def run_functional_checks(test_case, test_name, name, apply_fn, run_grad_checks,
                 'mean', 'inverse', 'triangular_solve', 'solve', 'addcmul',
                 'addcdiv', 'linalg.tensorinv', 'matrix_exp', 'qr',
                 'narrow', 'swapaxes', 'swapdims', 'tensor_split', 'tile',
-                'baddbmm', 'addbmm', 'addmv'] + separate_complex_tests
+                'baddbmm', 'addbmm', 'addmv', 'diag'] + separate_complex_tests
```

Comment:
we are trying to not add tests in `method_tests` anymore. could you please add an `OpInfo` based test for `diag`?